src/libraries/Microsoft.Extensions.Options/tests/SourceGenerationTests/ParserTests.cs

```diff
@@ -0,0 +1,931 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Options.Generators;
+using SourceGenerators.Tests;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Gen.OptionsValidation.Test;
+
+public partial class ParserTests
+{
+    [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBrowser))]
+    public async Task PotentiallyMissingAttributes()
+    {
+        var (d, _) = await RunGenerator(@"
```

Comment:
Consider moving tests like this one to the unit test project, so that any compiler errors during SG development don't impact our ability to run and debug these tests.